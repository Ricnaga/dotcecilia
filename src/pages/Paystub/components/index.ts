export * from './PaystubCalculator/PaystubCalculator';
export * from './PaystubTableBody/PaystubTableBody';
