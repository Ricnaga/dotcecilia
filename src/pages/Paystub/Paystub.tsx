import {
  PageContainer,
  PaymentContainer,
  PaymentTable,
} from '@shared/components';
import { PaystubCalculator, PaystubTableBody } from './components';
import { usePaystub } from './hooks/usePaystub';

export function Paystub() {
  const {
    data: { printRef, values },
    functions: { setPrint, onChangeValue, onCheckedValue },
  } = usePaystub();
  return (
    <PageContainer title="Holerite">
      <PaymentContainer
        calculator={
          <PaystubCalculator
            onPrint={setPrint}
            values={values}
            onChange={onChangeValue}
            onChecked={onCheckedValue}
          />
        }
        printRef={printRef}
      >
        <PaymentTable
          headerType="PAGAMENTO"
          name={values.name}
          monthRef={values.initialWorkdayMonth}
        >
          <PaystubTableBody values={values} />
        </PaymentTable>
      </PaymentContainer>
    </PageContainer>
  );
}
