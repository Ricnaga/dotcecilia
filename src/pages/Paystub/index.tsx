import React from 'react';
import { CeciliaPageHeader, CeciliaTable } from '../../shared/components';
import {
  PaystubCalculator,
  PaystubCalculatorFields,
  PaystubTableBody,
} from './components';

export function PaystubPage() {
  const [values, setValues] = React.useState<PaystubCalculatorFields>({
    name: '',
    hasUnsanitary: false,
    unsanitary: 0,
    extraHour: false,
    fullExtra: false,
    hours: 0,
    salary: 0,
    RefDate: new Date().toLocaleDateString('pt-br'),
    vtr: 0,
    missingDays: 0,
    discountedDays: 0,
  });
  console.log(values);
  const onChangeValue = (
    key: keyof PaystubCalculatorFields,
    value: number | string,
  ) =>
    value
      ? setValues({
          ...values,
          [key]: value,
        })
      : setValues({
          ...values,
          [key]: 0,
        });

  const onCheckedValue = (
    key: keyof Pick<
      PaystubCalculatorFields,
      'hasUnsanitary' | 'extraHour' | 'fullExtra'
    >,
  ) => setValues({ ...values, [key]: !values[key] });

  return (
    <CeciliaPageHeader title="Holerite">
      <div className="grid grid-cols-12 gap-10">
        <div className="grid grid-cols-12 col-span-7">
          <CeciliaTable headerType="PAGAMENTO">
            <PaystubTableBody />
          </CeciliaTable>
        </div>
        <div className="col-span-5">
          <PaystubCalculator
            values={values}
            onChange={onChangeValue}
            onChecked={onCheckedValue}
          />
        </div>
      </div>
    </CeciliaPageHeader>
  );
}
