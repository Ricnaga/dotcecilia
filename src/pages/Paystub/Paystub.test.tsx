import { render } from '@testing-library/react';
import { PaystubPage } from './Paystub';

describe('Page: PaystubPage', () => {
  it('should render correctly', () => {
    const { container } = render(<PaystubPage />);
    expect(container).toMatchSnapshot();
  });
});
