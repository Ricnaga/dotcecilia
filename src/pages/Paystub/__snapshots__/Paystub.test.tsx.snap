// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Paystub should render correctly 1`] = `
<div>
  <div
    class="w-full"
  >
    <p
      class="text-5xl font-bold my-6"
    >
      Holerite
    </p>
    <div
      class="grid grid-cols-12 gap-10"
    >
      <div
        class="grid grid-cols-12 col-span-7"
      >
        <table
          class="col-span-12 h-1 bg-slate-100 border border-slate-800 text-slate-900"
        >
          <thead
            class="bg-yellow-100"
          >
            <tr>
              <th
                colspan="6"
              >
                undefined
              </th>
              <th
                colspan="2"
              >
                RECIBO DE 
                PAGAMENTO
              </th>
            </tr>
          </thead>
          <thead>
            <tr>
              <th
                class="text-right"
                colspan="6"
              >
                undefined
              </th>
              <th
                colspan="2"
              >
                Ref. OUTUBRO
              </th>
            </tr>
            <tr>
              <th
                colspan="8"
              >
                CNPJ: 
                undefined
              </th>
            </tr>
            <tr>
              <th
                colspan="8"
              >
                <p
                  class="my-4 border-y border-slate-800"
                >
                  Nome
                </p>
              </th>
            </tr>
          </thead>
          <tbody
            class="border-t border-slate-800"
          >
            <tr>
              <td
                class="border-r border-slate-800 undefined"
                colspan="3"
              />
              <td
                class="border-r border-slate-800 text-center"
                colspan="1"
              >
                Ref.
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                Vencimento
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              >
                Descontos
              </td>
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                class="border-r border-slate-800 undefined"
                colspan="3"
              >
                Salário do mês
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="1"
              >
                30
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                 0,00
              </td>
              <td
                class="border-r border-slate-800 border-b"
                colspan="2"
              />
            </tr>
            <tr
              class="border-t border-slate-800 border"
            >
              <td
                class="border-r border-slate-800 undefined"
                colspan="2"
              >
                Vtr - 
                (08 à 08/11/2022)
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                 0,00
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                6
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                 0,00
              </td>
              <td
                class="border-r border-slate-800 border-r"
                colspan="2"
              />
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                class="border-r border-slate-800 undefined"
                colspan="3"
              >
                Falta dia
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                0
              </td>
              <td
                class="border-r border-slate-800 text-center"
              />
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              >
                 0,00
              </td>
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                class="border-r border-slate-800 undefined"
                colspan="3"
              >
                Adiantamento anterior
              </td>
              <td
                class="border-r border-slate-800 text-center"
              />
              <td
                class="border-r border-slate-800 text-center"
              />
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              >
                 0,00
              </td>
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                class="border-r border-slate-800 undefined"
                colspan="3"
              >
                Desconto 6% Vtr
              </td>
              <td
                class="border-r border-slate-800 text-center"
              />
              <td
                class="border-r border-slate-800 text-center"
              />
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              >
                 0,00
              </td>
            </tr>
            <tr
              class="border-t border-slate-800"
            >
              <td
                colspan="4"
              >
                <p
                  class="mt-4 border-r border-t border-slate-800"
                >
                  Total:
                </p>
              </td>
              <td
                colspan="1"
              >
                <p
                  class="mt-4 text-center border-r border-t border-slate-800"
                >
                   0,00
                </p>
              </td>
              <td
                colspan="3"
              >
                <p
                  class="mt-4 text-center border-t border-slate-800"
                >
                   0,00
                </p>
              </td>
            </tr>
            <tr
              class="mb-4 border-y border-slate-800"
            >
              <td
                colspan="4"
              >
                <p
                  class="mb-4 border-b border-r border-slate-800"
                >
                  Total líquido:
                </p>
              </td>
              <td
                colspan="4"
              >
                <p
                  class="mb-4 border-b border-slate-800 text-center"
                >
                   0,00
                </p>
              </td>
            </tr>
          </tbody>
          <tfoot
            class="border-t border-slate-800"
          >
            <tr>
              <td
                colspan="4"
              >
                <p
                  class="my-8 text-center"
                >
                  Assinatura:.......................................................
                </p>
              </td>
              <td
                colspan="4"
              >
                <p
                  class="my-8 text-center"
                >
                  Data:__/__/__
                </p>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
      <div
        class="col-span-5"
      >
        <div
          class="grid grid-cols-12"
        >
          <p
            class="text-4xl text-left font-bold mb-4 col-span-6"
          >
            Cálculos
          </p>
          <button
            class="btn btn-yellow col-start-9 col-end-13 mb-4"
            type="button"
          >
            Imprimir
          </button>
        </div>
        <div
          class="grid grid-cols-12 gap-4"
        >
          <div
            class="col-span-12"
          >
            <label
              class="text-xl"
              for="text"
            >
              <input
                placeholder="Digite o nome"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="col-span-6"
          >
            <div
              class="flex items-center"
            >
              <label
                class="mr-4"
                id="headlessui-label-:r0:"
              >
                Insalubridade ?
              </label>
              <button
                aria-checked="false"
                aria-labelledby="headlessui-label-:r0:"
                class="btn-switch false "
                data-headlessui-state=""
                id="headlessui-switch-:r1:"
                role="switch"
                tabindex="0"
                type="button"
              >
                <span
                  class="translate-x-1 inline-block h-4 w-4 transform rounded-full bg-white transition-transform"
                />
              </button>
            </div>
          </div>
          <div
            class="col-span-6 grid grid-cols-12 justify-around"
          >
            <div
              class="col-span-6"
            >
              <div
                class="flex items-center"
              >
                <label
                  class="mr-4"
                  id="headlessui-label-:r2:"
                >
                  HE?
                </label>
                <button
                  aria-checked="false"
                  aria-labelledby="headlessui-label-:r2:"
                  class="btn-switch false "
                  data-headlessui-state=""
                  id="headlessui-switch-:r3:"
                  role="switch"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="translate-x-1 inline-block h-4 w-4 transform rounded-full bg-white transition-transform"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="col-span-6"
          >
            <label
              class="text-2xl"
              for="number"
            >
              <input
                min="0.00"
                placeholder="salário"
                step="0.01"
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="col-span-6"
          >
            <label
              class="text-2xl"
              for="number"
            >
              <input
                min="0"
                placeholder="Dias faltados"
                step="1"
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="col-span-6"
          >
            <label
              class="text-2xl"
              for="number"
            >
              <input
                min="0.00"
                placeholder="Valor diário VTR"
                step="0.01"
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="col-span-6"
          >
            <label
              class="text-2xl"
              for="number"
            >
              <input
                min="0"
                placeholder="Desconto de dias"
                step="1"
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="col-span-6"
          >
            <label
              class="text-xl"
              for="date"
            >
              Primeiro dia útil
              <input
                type="date"
                value="2022-11-08"
              />
            </label>
          </div>
          <div
            class="col-span-6"
          >
            <label
              class="text-xl"
              for="date"
            >
              Último dia útil
              <input
                type="date"
                value="2022-11-08"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
